src/api/test/functional/webui/user_controller_test.rb

```diff
@@ -181,6 +181,22 @@ def test_that_redirect_after_login_works # spec/controllers/webui/users_spec.rb
     assert_equal search_path, current_path
   end
 
+  def test_that_redirect_from_user_do_login_works # spec/controllers/webui/users_spec.rb
+    use_js
+
+    visit search_path
+    click_link("Log In")
+    fill_in 'Username', with: "tom"
+    fill_in 'Password', with: "brokenpassword"
+    click_button 'Log In'
```

Comment:
You could use "/user/login", but if there is a named route we usually use them. In this case it would be user_login_path.

That's how you can get information about routes in a rails app: ```rake routes | grep login```